keps/sig-auth/4193-bound-service-account-token-improvements/README.md

```diff
@@ -0,0 +1,614 @@
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+* `ServiceAccountTokenJTI` feature flag will toggle including JTI information in tokens, as well as recording JTIs in the audit log / the SA user info.
+* `ServiceAccountTokenPodNodeInfo` feature flag will toggle including node info associated with pods in tokens.
+* `ServiceAccountTokenNodeBinding` feature flag will toggle allowing service account tokens to be bound to Node objects.
+
+Both of these feature flags can be disabled without any unexpected adverse affects or coordination required.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate
+  - Feature gate name: `ServiceAccountTokenJTI`
+  - Components depending on the feature gate: kube-apiserver
+
+- [x] Feature gate
+  - Feature gate name: `ServiceAccountTokenPodNodeInfo`
+  - Components depending on the feature gate: kube-apiserver
+
+- [x] Feature gate
+  - Feature gate name: `ServiceAccountTokenNodeBinding`
+  - Components depending on the feature gate: kube-apiserver
+
+- [x] Feature gate
+  - Feature gate name: `ServiceAccountTokenNodeBindingValidation`
+  - Components depending on the feature gate: kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature gate will cause additional information to be stored/persisted into service account JWTs, as well
+as new audit annotations being recorded in the audit log. This is all purely additive, so no changes to existing
+features, schemas or fields are expected.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Future tokens will then not embed this information. Any existing issued tokens **will** still have this
+information embedded however.
+
+If these fields are deemed to be problematic for other systems interpreting these tokens, users will need to re-issue
+these tokens before presenting them elsewhere.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Future tokens will once again include this information/no adverse effects.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes (as noted above in the test plan)
+
+### Rollout, Upgrade and Rollback Planning
+
+Rolling this out will be done by enabling the feature flag on all control plane hosts.
+This should not have any issues/affect during upgrades.
+Rollback is done by removing/disabling the feature gate.
```

Comment:
This should note that the `ServiceAccountTokenNodeBindingValidation` feature gate should be enabled and complete rollout before the `ServiceAccountTokenNodeBinding` gate is enabled, so all active servers will correctly validate tokens issued by any server.

It should also note that we would default `ServiceAccountTokenNodeBindingValidation` on one release before defaulting `ServiceAccountTokenNodeBinding` on.